### ISSUE TITLE
TS: Need exported flatbuffer files to use generic dist bundle

### ISSUE THREAD

**Issue by kylebarron** [bjornharrtell/flatgeobuf#70]
A bit of a continuation of https://github.com/bjornharrtell/flatgeobuf/issues/67... 

I'd like to express that it's difficult currently to use the generic `dist` bundle (`flatgeobuf.min.js`), because the only exports are `serialize` and `deserialize`. 

As described earlier in #67, with the generic endpoint, you have to write your own function to parse each feature. As in the GeoJSON version, you'd have to parse the geometries and properties using something similar to `parseProperties` and `fromGeometry`:
https://github.com/bjornharrtell/flatgeobuf/blob/170a59153fe19cbf163fc296c75cf132461a456f/src/ts/geojson/feature.ts#L18-L19

But in order to parse geometries, you need to know the `GeometryType`:
https://github.com/bjornharrtell/flatgeobuf/blob/170a59153fe19cbf163fc296c75cf132461a456f/src/ts/geojson/geometry.ts#L74-L86

And in order to parse properties, you need to know the `ColumnType`:
https://github.com/bjornharrtell/flatgeobuf/blob/170a59153fe19cbf163fc296c75cf132461a456f/src/ts/generic/feature.ts#L116-L125

Therefore, `flatgeobuf.min.js` is rather useless, because you still need to copy all the generated flatbuffer files to do anything.

I'm unable to use anything in `lib/` because it's not compiled at all, and I get constant 
```
Uncaught SyntaxError: Cannot use import statement outside a module
```
errors in Node. (We're working on [loaders.gl](https://loaders.gl), a collection of JS file loaders that need to work in both browser and Node.)

In summary, I'd suggest exporting more objects in the generic module. The `ol` and `geojson` dist bundles make sense to only have `deserialize` and `serialize` exports because there's no customization, but I think at a minimum you should re-export the generated flatbuffer files.

**Comment by kylebarron** [comment]
I think for now I'll just copy the generated flatbuffer TS files into our project, and that should unblock me on parsing to flat arrays (i.e. #51)

**Comment by bjornharrtell** [comment]
You should be able to use /lib in node if you have module support either native or via esm.

**Comment by bjornharrtell** [comment]
Closing as I don't know if this is relevant any longer, please reopen if so.

**Comment by kylebarron** [comment]
I haven't tested it yet, but there should be a way in our build system to use `esm`.

Regardless though, the `generic` dist bundle is still unusable because it exposes the flatbuffer directly, and that requires the flatbuffer definitions to parse.

**Comment by bjornharrtell** [comment]
I agree, I will see if I can export what's needed into the generic dist.

**Comment by kylebarron** [comment]
It looks like https://github.com/bjornharrtell/flatgeobuf/blob/master/src/ts/generic.ts is the index file for the generic bundle. 

For my custom parser I needed the `GeometryType` and `ColumnType` enums from `header_generated.ts` (Not sure why `ColumnType` is duplicated in `ColumnType.ts`).

**Comment by bjornharrtell** [comment]
Exports done with https://github.com/bjornharrtell/flatgeobuf/commit/20fe11126a67e58af2e5194d70000a4d1a40d525 and released as 3.5.2. Looks correct but I haven't verified 100% so let me know if it doesn't work as expected. I may be able to add more examples that also uses the generic dist at a later time.
